README.md

```diff
@@ -0,0 +1,123 @@
+meta-rust-bin
+=============
+
+An OpenEmebdded/Yocto layer providing pre-built toolchains for the
+[Rust programming language](https://www.rust-lang.org).
+
+<!-- toc -->
+
+- [Basic Example](#basic-example)
+- [Features](#features)
+- [Advanced Features](#advanced-features)
+  * [Specifying Cargo Features](#specifying-cargo-features)
+  * [Using Components Individually](#using-components-individually)
+- [Pre-built vs. Compiled](#pre-built-vs-compiled)
+- [Copyright](#copyright)
+
+<!-- tocstop -->
+
+
+## Basic Example
+
+A basic class for cargo-based executables is provided. The following is a
+simple recipe that builds the [gpio-utils](https://github.com/rust-embedded/gpio-utils)
+crate from a branch tagged with the version `${PV}`:
+
+```bitbake
+inherit cargo
+
+SUMMARY = "GPIO Utilities"
+HOMEPAGE = "git://github.com/rust-embedded/gpio-utils"
+LICENSE = "MIT"
+
+SRC_URI = "https://github.com/rust-embedded/gpio-utils.git;tag=${PV}"
+S = "${WORKDIR}/git"
+
+LIC_FILES_CHKSUM = "file://LICENSE-MIT;md5=935a9b2a57ae70704d8125b9c0e39059"
+```
+
+As you can see, there is almost no overhead introduced from the `cargo` class
+beyond simply inheriting it. The `cargo` class adds the appropriate Rust
+dependencies as well as default compile and install steps.
+
+
+## Features
+
+Currently supported:
+
+  * Rust 1.11
+  * x86 (32 and 64-bit), ARM (32 and 64-bit) build systems.
+  * All Linux architectures that Rust itself supports (Multiple flavors of:
+    x86, ARM, PPC, and MIPS)
+
```

Comment:
One other question that this raised for me was making sure that all system libraries needed by libstd were getting installed.

Here's a dump of `objdump -x gpio | grep NEEDED`:

```
  NEEDED               libdl.so.2
  NEEDED               librt.so.1
  NEEDED               libpthread.so.0
  NEEDED               libgcc_s.so.1
  NEEDED               libc.so.6
  NEEDED               ld-linux-armhf.so.3
  NEEDED               libm.so.6
```

This matches well with what I would expect from the [Rust book linking section](https://doc.rust-lang.org/book/advanced-linking.html#linux).

Yocto adds a C compiler and libc to the `DEPENDS` of a package by default through an undocumented variable `BASEDEPENDS` (so long as [INHIBIT_DEFAULT_DEPS](https://www.yoctoproject.org/docs/2.1/ref-manual/ref-manual.html#var-INHIBIT_DEFAULT_DEPS) isn't specified). On my system `BASEDEPENDS` is

```
BASEDEPENDS="virtual/arm-poky-linux-gnueabi-gcc virtual/arm-poky-linux-gnueabi-compilerlibs virtual/libc"
```

so it looks like we should get everything we need so long as we don't specify `INHIBIT_DEFAULT_DEPS`.

I'd say this isn't completely air-tight, since the `BASEDEPENDS` could be something else in other systems. But for now I'm happy running with what we've got and fixing bugs if they crop up.